ixwebsocket/IXSocketConnect.cpp

```diff
@@ -148,4 +160,34 @@ namespace ix
         setsockopt(sockfd, SOL_SOCKET, SO_NOSIGPIPE, (void*) &value, sizeof(value));
 #endif
     }
+    int SocketConnect::connectToAddressViaProxy(const std::string& host,
+                                                int port,
+                                                std::string& errMsg, const ProxySetup &proxy_settings)
+    {
+        char* errorMsg;
+
+
+        //        auto proxyConfig = ix::make_unique<proxysocketconfig>(proxysocketconfig_create(proxy_settings.get_proxy_type(), proxy_settings.get_proxy_host().c_str(),
+        //                                                                                          proxy_settings.get_proxy_port(), proxy_settings.get_proxy_user().c_str(),
+        //                                                                                          proxy_settings.get_proxy_pass().c_str()), [] (proxysocketconfig_struct* obj){proxysocketconfig_free(obj);});
+
+        //        std::unique_ptr<proxysocketconfig_struct, decltype(&proxysocketconfig_free)> proxyConfig(proxysocketconfig_create(proxy_settings.get_proxy_type(), proxy_settings.get_proxy_host().c_str(),
+        //                                                                                                                         proxy_settings.get_proxy_port(), proxy_settings.get_proxy_user().c_str(),
+        //                                                                                                                         proxy_settings.get_proxy_pass().c_str()), &proxysocketconfig_free);
+        auto proxyConfig = proxysocketconfig_create(proxy_settings.get_proxy_type(), proxy_settings.get_proxy_host().c_str(),
+                                                    proxy_settings.get_proxy_port(), proxy_settings.get_proxy_user().c_str(),
+                                                    proxy_settings.get_proxy_pass().c_str());
+        proxysocketconfig_set_logging(proxyConfig, logger, nullptr);
+        int fd = proxysocket_connect(proxyConfig, host.c_str(), port, &errorMsg);
+
+        if (fd == -1)
+        {
+            errMsg = errorMsg;
+            return -1;
+        }
+        proxysocketconfig_free(proxyConfig);
```

Comment:
If the previous condition is true (`fd == -1`), then you'll get a memory leak because `free` will be not called.
Use std::unique_ptr.